operator/transformer/recombine/recombine.go

```diff
@@ -231,10 +232,16 @@ func (r *RecombineOperator) Process(ctx context.Context, e *entry.Entry) error {
 		// Add the current log to the new batch
 		r.addToBatch(ctx, e, s)
 		return nil
-	}
-
 	// This is the last entry in a complete batch
-	if matches && r.matchIndicatesLast() {
+	case matches && r.matchIndicatesLast():
+		r.addToBatch(ctx, e, s)
+		err := r.flushSource(s)
+		if err != nil {
+			return err
+		}
+		return nil
+	// When matching on first entry, never batch partial first. Just emit immediately
+	case !matches && r.matchIndicatesFirst() && len(r.batchMap[s]) == 0:
```

Comment:
This case wound up with the exact same logic as the one above. I think in this case though, we do not need to mess with batching and flushing, and instead can just do `r.Write(context.Background(), entry)`.

I don't feel strongly about this, but at least we should deduplicate this code. Could just be:
```suggestion
	case matches && r.matchIndicatesLast():
		fallthrough
	// When matching on first entry, never batch partial first. Just emit immediately
	case !matches && r.matchIndicatesFirst() && len(r.batchMap[s]) == 0:
		r.addToBatch(ctx, e, s)
		return r.flushSource(s)
```